selftests/functional/test_statuses.py

```diff
@@ -0,0 +1,291 @@
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+from avocado.utils import process
+from avocado.utils import script
+
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO = os.environ.get("UNITTEST_AVOCADO_CMD", "./scripts/avocado")
+
+TEST_CONTENT = """
+import os
+import sys
+from avocado import Test
+
+class StatusTest(Test):
+
+    def setUp(self):
+        self.log.info('setup pre')
+        if self.params.get('location') == 'setUp':
+            exec(self.params.get('command'))
+        self.log.info('setup post')
+
+    def test(self):
+        self.log.info('test pre')
+        if self.params.get('location') == 'test':
+            exec(self.params.get('command'))
+        self.log.info('test post')
+
+    def tearDown(self):
+        self.log.info('teardown pre')
+        if self.params.get('location') == 'tearDown':
+            exec(self.params.get('command'))
+        self.log.info('teardown post')
+"""
+
+YAML_CONTENT = """
+locations: !mux
+    setup:
+        location: setUp
+    test:
+        location: test
+    teardown:
+        location: tearDown
+
+commands: !mux
+    skip:
+        command: self.skip()
+    cancel:
+        command: self.cancel()
+    fail:
+        command: self.fail()
+    warn:
+        command: self.log.warn('')
+    exception:
+        command: raise Exception
+    exit:
+        command: sys.exit(-1)
+    kill:
+        command: os.kill(os.getpid(), 9)
+"""
+
+# Format: {variant1: (status,
+#                    [msg_in, ...],
+#                    [msg_not_in, ...]),
+#          variant2: (status,
+#                     [msg_in, ...],
+#                     [msg_not_in, ...])}
+EXPECTED_RESULTS = {'skip-setup-d304': ('SKIP',
+                                        ['setup pre',
+                                         'teardown pre',
+                                         'teardown post',
+                                         "[WARNING: self.skip() will be "
+                                         "deprecated. Use 'self.cancel()' "
+                                         "or the skip decorators]"],
+                                        ['setup post',
+                                         'test pre',
+                                         'test post']),
```

Comment:
The second comment feels like http://devopsreactions.tumblr.com/post/150439088410/overengineering :-)
I will follow the suggestion in the first comment though. It will remove some code keeping the readability. Thank you.